### ISSUE TITLE
References and Borrowing section uses rust specific definition of path prior to defining it.

### ISSUE THREAD

**Issue by rs333** [cognitive-engineering-lab/rust-book#176]
- I have searched open and closed issues and pull requests for duplicates, using these search terms:
  - path
  - paths
- I have checked the latest `main` branch to see if this has already been fixed, in this file:
  - `ch04-02-references-and-borrowing.html`

URL to the section(s) of the book with this problem:
  `ch04-02-references-and-borrowing.html`

Description of the problem:
 The revised version of `ch04-02-references-and-borrowing.html` introduces the term Paths to mean 
 ```
 More generally, permissions are defined on paths and not just variables. A path is anything you can put on the left-hand side of an assignment. Paths include:

Variables, like a.
Dereferences of paths, like *a.
Array accesses of paths, like a[0].
Fields of paths, like a.0 for tuples or a.field for structs (discussed next chapter).
Any combination of the above, like *((*a)[0].1).
```

This seems to be a rust specific overload of the common meaning to the term path.  In its present form, it can cause cognitive dissidence for the reader. (At least one and probably more :)   Additionally, it appears to be technically not correct because while some paths can be put on the left hand side of an assignment, not all paths can be put on the left hand side of an assignment.  E.g. I don't believe a path referring to a module item can be on the left hand side of an assignment.

Suggested fix:
1. Define path prior to it's use using the definition from the Rust [Reference](https://doc.rust-lang.org/reference/paths.html)

A path is a sequence of one or more path segments logically separated by a 
namespace qualifier (::). If a path consists of only one segment, it refers to 
either an [item](https://doc.rust-lang.org/reference/items.html) or a [variable](https://doc.rust-lang.org/reference/variables.html) in a local control scope. If a path has multiple 
segments, it always refers to an item.

3. Consider adding a modifier to the term path in this section to make it clear which type they are referring to.  E.g. `Assignable paths`.
4. In the text shown int he Description of the Problem, revise the text to indicate that permissions are defined on assignable paths which are paths that can be on the left-hand side of an assignment.    Something like
```
More generally, permissions are defined on assignable paths which are paths that can be put on the left-hand side of an assignment.  Examples of assignable paths include:

Variables, like a.
Dereferences of assignable paths, like *a.
Array accesses of assignable paths, like a[0].
Fields of assignable paths, like a.0 for tuples or a.field for structs (discussed next chapter).
Any combination of the above, like *((*a)[0].1).
```

That said, given this [place-expressions-and-value-expresions](https://doc.rust-lang.org/stable/reference/expressions.html#place-expressions-and-value-expressions) text from the Rust Reference, it may be more appropriate to call it a place expression vice a path.

**Comment by willcrichton** [comment]
Hi, this is a good point. These two meanings of path are conflicting. I'll note that the borrow-checker-oriented meaning of "path" is based on @nikomatsakis's borrow checker formalizations. I'm going to discuss with him what he recommends for terminology here.

**Comment by willcrichton** [comment]
I have addressed this in #211. I decided to just go for "place" instead of explaining the distinction between places vs. place expressions for simplicity.
